django/thunderstore/repository/comment.py

```diff
@@ -0,0 +1,85 @@
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import PermissionDenied
+
+from thunderstore.community.models import PackageListing
+from thunderstore.core.utils import capture_exception
+from thunderstore.repository.models import Comment
+
+
+def clean_content(content: str) -> str:
+    return content.strip()
+
+
+class CreateCommentForm(forms.ModelForm):
+    class Meta:
+        model = Comment
+        fields = ["content"]
+
+    def __init__(
+        self,
+        user: User,
+        commented_object: PackageListing,
+        *args,
+        **kwargs,
+    ) -> None:
+        if not isinstance(commented_object, PackageListing):
+            raise ValueError("Unsupported parent object for comment")
+
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.commented_object = commented_object
+
+    def clean_content(self) -> str:
+        return clean_content(self.cleaned_data["content"])
+
+    def save(self, *args, **kwargs) -> Comment:
+        self.instance.author = self.user
+        self.instance.thread = self.commented_object
+        return super().save(*args, **kwargs)
+
+
+class EditCommentForm(forms.Form):
+    content = forms.CharField(max_length=Comment._meta.get_field("content").max_length)
+    is_pinned = forms.BooleanField(required=False)
+
+    def __init__(self, user: User, comment: Comment, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.comment = comment
+
+    def can_edit_content(self) -> bool:
```

Comment:
This permission check probably should be moved out of the form (e.g. to the model, akin to how UploaderIdentity has permission checks related to that now)